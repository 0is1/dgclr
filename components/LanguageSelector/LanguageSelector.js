import React, { PureComponent } from 'react';
import { i18n } from 'i18n';
import Styles from './LanguageSelector.styles';

const { Text } = Styles;

type Props = {
  currentLanguage: ?string,
  language: string,
};

class LanguageSelector extends PureComponent<Props> {
  changeLanguage = async () => {
    const { language } = this.props;
    try {
      await i18n.changeLanguage(language);
    } catch (e) {
      console.log(e);
    }
  };

  render() {
    const { currentLanguage, language } = this.props;
    const languageToCompare = i18n.language || currentLanguage;
    return (
      <Text active={languageToCompare === language} onClick={this.changeLanguage}>
        {language}
      </Text>
    );
  }
}

export default LanguageSelector;
