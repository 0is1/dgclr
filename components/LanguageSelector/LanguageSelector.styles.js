import styled from 'styled-components';
import colors from 'components/colors';

const Styles = {
  Wrapper: styled.div`
    align-items: center;
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    width: 60%;
  `,
  Text: styled.a`
    cursor: pointer;
    font-size: 1rem;
    border-bottom: ${(props) => (props.active ? `2px solid ${colors.hover}` : 'none')};
    margin-right: 1.25rem;
    padding: 0 .25rem .25rem;
    position: relative;
    :not(:last-of-type) {
        :after {
            content: '/';
            color: ${colors.darkGray};
            position: absolute;
            right: -.75rem;
            top: 0;
        }
      }
    }
  `,
};

export default Styles;
