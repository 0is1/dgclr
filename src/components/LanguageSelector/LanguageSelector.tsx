import React, { useEffect } from 'react';
import { useTranslation } from 'next-i18next';
import { RadioChangeEvent, Tooltip } from 'antd';
import { useRouter } from 'next/router';
import { Radio } from 'antd';
import useLocalStorageState from 'use-local-storage-state';

const options = [
  { label: 'ðŸ‡«ðŸ‡®', value: 'fi' },
  { label: 'ðŸ‡ºðŸ‡¸', value: 'en' },
];

const LanguageSelector = () => {
  const router = useRouter();
  const { i18n, t } = useTranslation();
  const [locale, setLocale] = useLocalStorageState('locale', {
    defaultValue: 'fi',
  });
  useEffect(() => {
    const toggleLang = async () => {
      await i18n.changeLanguage(locale);
    };
    if (typeof i18n?.changeLanguage === 'function') toggleLang();
  }, [i18n, locale]);

  return (
    <Radio.Group
      onChange={async ({ target: { value } }: RadioChangeEvent) => {
        setLocale(value);
        await i18n.changeLanguage(value);
        router.push(`${router.pathname}`, undefined, { locale: value });
      }}
      value={locale}
      optionType="button"
    >
      {options.map((option) => (
        <Tooltip title={t('common:change_language')} key={option.value}>
          <Radio.Button value={option.value}>{option.label}</Radio.Button>
        </Tooltip>
      ))}
    </Radio.Group>
  );
};

export default LanguageSelector;
