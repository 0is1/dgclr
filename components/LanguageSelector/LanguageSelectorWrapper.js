import React from 'react';
import { config } from 'lib/i18n';
import { isArrayWithLength } from 'helpers/utils';
import LanguageSelector from './LanguageSelector';
import Styles from './LanguageSelector.styles';

const { Wrapper } = Styles;

type Props = { currentLanguage: ?string };

function LanguageSelectorWrapper({ currentLanguage }: Props) {
  const languages = config.allLanguages;

  if (!isArrayWithLength(languages)) {
    return null;
  }
  const languageSelectors = languages.map((language) => (
    <LanguageSelector key={language} language={language} currentLanguage={currentLanguage} />
  ));
  return <Wrapper>{languageSelectors}</Wrapper>;
}

export default LanguageSelectorWrapper;
