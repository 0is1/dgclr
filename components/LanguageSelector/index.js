import LanguageSelectorWrapper from './LanguageSelectorWrapper';

export default LanguageSelectorWrapper;
