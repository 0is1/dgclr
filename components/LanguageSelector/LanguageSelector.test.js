import React from 'react';
import { shallow } from 'enzyme';
import LanguageSelector from './LanguageSelector';
import Styles from './LanguageSelector.styles';

const { Text } = Styles;

const mockProps = (props = {}) => ({
  currentLanguage: 'fi',
  language: 'fi',
  ...props,
});

describe('<LanguageSelector />', () => {
  describe('snapshots', () => {
    it('renders with default props', () => {
      const props = mockProps();
      const subject = shallow(<LanguageSelector {...props} />);
      const textComponent = subject.find(Text);
      expect(textComponent.prop('active')).toEqual(true);
      expect(textComponent.text()).toEqual('fi');
      expect(subject).toMatchSnapshot();
    });
    it('renders with language prop', () => {
      const props = mockProps({ language: 'en' });
      const subject = shallow(<LanguageSelector {...props} />);
      const textComponent = subject.find(Text);
      expect(textComponent.prop('active')).toEqual(false);
      expect(textComponent.text()).toEqual('en');
      expect(subject).toMatchSnapshot();
    });
  });
});
