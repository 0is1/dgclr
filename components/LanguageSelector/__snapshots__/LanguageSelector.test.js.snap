// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageSelector /> snapshots renders with default props 1`] = `
<LanguageSelectorstyles__Text
  active={true}
  onClick={[Function]}
>
  fi
</LanguageSelectorstyles__Text>
`;

exports[`<LanguageSelector /> snapshots renders with language prop 1`] = `
<LanguageSelectorstyles__Text
  active={false}
  onClick={[Function]}
>
  en
</LanguageSelectorstyles__Text>
`;
